system/lib/wasmfs/file.h

```diff
@@ -315,11 +323,17 @@ class File::Handle {
     file->mode = (file->mode & S_IFMT) | (mode & ~S_IFMT);
   }
   time_t getCTime() { return file->ctime; }
+  long getCTimeNs() { return file->ctime_nsec; }
   void setCTime(time_t time) { file->ctime = time; }
+  void setCTimeNs(long time) { file->ctime_nsec = time; }
   time_t getMTime() { return file->mtime; }
+  long getMTimeNs() { return file->mtime_nsec; }
   void setMTime(time_t time) { file->mtime = time; }
+  void setMTimeNs(long time) { file->mtime_nsec = time; }
   time_t getATime() { return file->atime; }
+  long getATimeNs() { return file->atime_nsec; }
   void setATime(time_t time) { file->atime = time; }
+  void setATimeNs(long time) { file->atime_nsec = time; }
```

Comment:
Maybe we can combine these getters and setters? I imagine that usually the seconds and nanoseconds will always be set or retrieved at the same time.

Reply:
I think I was worried about compatibility with any code that used the old getters/setters by getting just the seconds with `time(NULL)`. Would it be worth switching all of them to use a combined getter/setter? If so, would it be better to pass in the seconds and nanoseconds together as a pointer to a struct or two separate `time_t` and `long` values respectively?

Reply:
Yes, switching all the use sites would be fine. There shouldn't be many of them, and this is private API so this wouldn't be a breaking change. I would implement it by passing the `timeval` struct by value (rather than by pointer) since it is so small.